.github/workflows/haskell.yml

```
@@ -49,84 +50,22 @@ jobs:
-            libtool `
-            make
-
-     # this seems to break something. It _must_ come after the pacman setup
-     # above. It appears as if PATHEXT is set _after_ ghcup install ghc/cabal, and
-     # as such we'd need pacman.exe instead.
-    - name: Setup Haskell
-      run: |
-        # Use GHCUP to manage ghc/cabal
-        ghcup install ghc --set ${{ matrix.ghc }}
-        ghcup install cabal --set 3.6.2.0
-
-        ghc --version
-        cabal --version
-
-    - name: "WIN: fixup cabal config"
-      if: runner.os == 'Windows'
-      run: |
-        # make sure cabal knows about msys64, and mingw64 tools. Not clear why C:/cabal/config is empty
-        # and C:/cabal doesn't even exist.  The ghcup bootstrap file should have create it in the image:
-        # See https://github.com/haskell/ghcup-hs/blob/787edc17af4907dbc51c85e25c490edd8d68b80b/scripts/bootstrap/bootstrap-haskell#L591
-        # So we'll do it by hand here for now.
-        #
-        # We'll _not_ add extra-include-dirs, or extra-lib-dirs, and rely on what's shipped with GHC.
-        # https://github.com/msys2/MINGW-packages/issues/10837#issuecomment-1047105402
-        # https://gitlab.haskell.org/ghc/ghc/-/issues/21111
-        # if we _do_ want them, this would be the lines to add below
-
-        $ghcMingwDir = Join-Path -Path $(ghc --print-libdir) `
-                                 -ChildPath ../mingw/x86_64-*-mingw32/lib/ `
-                                 -Resolve
-
-        cabal user-config -a "extra-prog-path: C:/msys64/mingw64/bin, C:/msys64/usr/bin" `
-                          -a "extra-include-dirs: C:/msys64/mingw64/include" `
-                          -a ("extra-lib-dirs: {0}, C:/msys64/mingw64/lib" -f $ghcMingwDir) `
-                          -f init
-
-    # Unify the computation of the cabal store directory to a single step. This makes referencing the cabal
-    # store in later steps easier.
-    #
-    # We know on cabal store is located in the following locations by OS:
-    #
-    #   Linux: /home/runner/.cabal/store
-    #   MacOS: /Users/runner/.cabal/store
-    #   Windows: C:\cabal\store
-    #
-    # However, we compute these as best be can from information cabal provides in case this changes in the future.
-    - name: "Setup cabal-store"
-      id: cabal-store
-      shell: bash
-      run: |
-        cabal help user-config
-        cabal_config_file="$(cabal help user-config | grep -A 1 'You can edit the cabal configuration file to set defaults' | tail -n 1 | xargs)"
-
-        if [[ '${{ runner.os }}' != 'Windows' ]]; then
-          echo "cabal-store=$(dirname "$cabal_config_file")/store" | tee -a "$GITHUB_OUTPUT"
-        else
-          echo "cabal-store=C:\\cabal\\store" | tee -a "$GITHUB_OUTPUT"
-        fi
+    - name: Install Haskell
+      uses: input-output-hk/setup-haskell@v1
+      id: setup-haskell
+      with:
+        ghc-version: ${{ matrix.ghc }}
+        cabal-version: ${{ matrix.cabal }}
```

It is safe to hard code it here, instead of using a variable. Also using the latest version makes more sense, unless there is some reason we are forced to use 3.6.2?
```suggestion
        cabal-version: "3.8.1.0"
```

In fact, I wouldn't even oppose removing the setting and just using the latest cabal. I think it makes sense to see early if our tooling breaks upon a cabal upgrade

---

Using `cabal-version: latest`